htdocs/google/admin/google_contactsync.php

```diff
@@ -383,14 +393,24 @@
 			$i++;
 		}
 
-		$result=0;
-		if (count($gContacts)) $result=insertGContactsEntries($gdata, $gContacts, $objectstatic);
+		$resultEntries=0;
+		if (count($gContacts)) $resultEntries=insertGContactsEntries($gdata, $gContacts, $objectstatic);
 
-		if (is_numeric($result) && $result >= 0) {
+		if (is_numeric($resultEntries) && $resultEntries >= 0) {
 			$mesg = $langs->trans("PushToGoogleSucess", count($gContacts));
 		} else {
 			$error++;
-			$errors[] = $langs->trans("Error").' '.$result;
+			$errors[] = $langs->trans("Error").' '.$resultEntries;
+		}
+
+		$resultTags=0;
+		if (count($gContacts)) $resultTags=updateGContactGroups($gdata, $gContacts, 'contact');
+
+		if (is_numeric($resultTags) && $resultTags >= 0) {
+			$mesg .= '<br>'.$langs->trans("TagsCreatedSuccess");
+		} else {
+			$error++;
+			$errors[] = $langs->trans("Error").' '.$resultTags;
 		}
```

Comment:
Here you update the contact group even if the contact as not been created. Is that what you wanted to do ? And if yes why ?

Reply:
You are right, I think I can add the condition "if(!$error) {" L405

Reply:
Another solution would be, for each contact, to update its labels right after its creation, which suggests removing the 'updateGContactGroups' function, what do you think?

Reply:
I think the actual solution with a if(!$error) will work